addons/account_facturx/models/ir_actions_report.py

```diff
@@ -15,18 +11,26 @@ def _post_pdf(self, save_in_attachment, pdf_content=None, res_ids=None):
         if self.model == 'account.move' and res_ids and len(res_ids) == 1:
             invoice = self.env['account.move'].browse(res_ids)
             if invoice.is_sale_document() and invoice.state != 'draft':
-                xml_content = invoice._export_as_facturx_xml()
+                writer = OdooPdfFileWriter(pdf_content)
+                writer.make_pdfa(invoice.name)
+                self._add_facturx_attachments(writer, invoice)
+                pdf_content = writer.save_content()
 
-                # Add attachment.
-                reader_buffer = io.BytesIO(pdf_content)
-                reader = PdfFileReader(reader_buffer)
-                writer = PdfFileWriter()
-                writer.cloneReaderDocumentRoot(reader)
-                writer.addAttachment('factur-x.xml', xml_content)
-                buffer = io.BytesIO()
-                writer.write(buffer)
-                pdf_content = buffer.getvalue()
-
-                reader_buffer.close()
-                buffer.close()
         return super(IrActionsReport, self)._post_pdf(save_in_attachment, pdf_content=pdf_content, res_ids=res_ids)
+
+    def _add_facturx_attachments(self, writer, invoice):
+        """
+        Add the factur-x xml and xmp shema to the PDF.
+        """
+        xml_content = invoice._export_as_facturx_xml()
+        metadata = """<?xpacket begin="﻿" id="W5M0MpCehiHzreSzNTczkc9d"?>""".encode()
```

Comment:
Empty `begin` attribute: is it needed?

Reply:
Are you sure `id` is a static value? It looks weird to me

Reply:
see: [adobe xmp doc](https://wwwimages2.adobe.com/content/dam/acom/en/devnet/xmp/pdfs/XMP%20SDK%20Release%20cc-2016-08/XMPSpecificationPart3.pdf) page 41
In an older version of the same document, it was explicitly said to use that id. I also compared with other valid factur-x pdf documents and they all had the exact same id.

Reply:
As for the begin="", it is not empty, although it could be:  

> This required attribute indicates the beginning of a new packet. Its value is the Unicode zero-width nonbreaking space character U+FEFF, in the appropriate encoding (UTF-8, UTF-16, or UTF-32). It serves as a
byte-order marker, where the character is written in the natural order of the application (consistent with the byte
order of the XML data encoding).
For backward compatibility with earlier versions of the XMP packet specification, the value of this attribute can
be the empty string, indicating UTF-8 encoding.